imgparse/rotations.py

```diff
@@ -0,0 +1,78 @@
+"""Math used to handle gimbal lock/singularity."""
+
+from collections import namedtuple
+from math import asin, atan2, cos, pi, sin
+
+Quaternion = namedtuple("Quaternion", "q0 q1 q2 q3")
+
+
+def euler_to_quaternion(roll, pitch, yaw):
+    """Convert Euler angles to quaternions."""
+    # Intermediate calculations for 'cos(x/2)' of Euler angles.
+    cos_y = cos((yaw * pi) / 360.0)
+    cos_p = cos((pitch * pi) / 360.0)
+    cos_r = cos((roll * pi) / 360.0)
+
+    # Intermediate calculations for 'sin(x/2)' of Euler angles.
+    sin_y = sin((yaw * pi) / 360.0)
+    sin_p = sin((pitch * pi) / 360.0)
+    sin_r = sin((roll * pi) / 360.0)
+
+    q0 = cos_r * cos_p * cos_y + sin_r * sin_p * sin_y
+    q1 = sin_r * cos_p * cos_y - cos_r * sin_p * sin_y
+    q2 = cos_r * sin_p * cos_y + sin_r * cos_p * sin_y
+    q3 = cos_r * cos_p * sin_y - sin_r * sin_p * cos_y
+
+    return Quaternion(q0, q1, q2, q3)
+
+
+def quaternion_to_euler(q):
+    """Convert quaternions to Euler angles."""
+    # Compute pitch sine term.
+    sin_p = 2.0 * (q.q0 * q.q2 - q.q3 * q.q1)
+
+    # Pitch >= 89.0 deg ( gimbal lock condition ) ?
+    if sin_p >= 0.99985:
```

Comment:
This condition might read better as an `epsilon=0.00015` in the function definition and `if sin_p >= 1 - epsilon` respectively. An `epsilon_deg` in terms of degrees may be even better in that regard.

Reply:
Added `epsilon_deg`